src/api/use-subscription.ts

```diff
@@ -40,25 +38,29 @@ export function useSubscription({
         onMessage && onMessage(event);
       };
 
-      webSocket.onerror = (event) => {
-        setStatus(getSubscriptionStatus(webSocket.readyState));
-        Bugsnag.notify(`WebSocket error "${event.message}"`);
-        onError && onError(event);
-      };
```

Comment:
Not sure about removing this whole `onerror` handling. Are we sure it won't happen in production?

Reply:
It does happen in production, but when it happens it seems to me like it's not actual errors that we want to handle. It seems to happen on reconnects on android, otherwise it seems to be on network errors, which we are handling with onClose events.

https://app.bugsnag.com/atb-as/mittatb/errors/648adc57dd45910007005d13?filters%5Bevent.since%5D=30d&filters%5Bsearch%5D=websocket

The `event` value is also completely different on iOS and Android. Android is usually `null`, and on iOS it's some text in Norwegian. This makes them hard to work with, and to me it sounds like it's not made to be handled. We could keep some bugsnag logging here, but I think it's better to not have the onError callback, that sounds like it solves some problem it in reality doesn't.

Makes sense?

Reply:
One solution would be to keep something like this:

```ts
webSocket.onerror = (event) => {
  setStatus(getSubscriptionStatus(webSocket.readyState));
  if (event.message !== null) {
    Bugsnag.notify(`WebSocket error "${event.message}"`);
  }
};
```

Reply:
@gorandalum This seems to be working well locally, so updated now. Does that look ok?

Reply:
I'll read a little documentation first, so approval probably incoming soon :+1:

Just a note, I think the `useSubscription` hook shouldn't take in an `onError` if it isn't used.

Reply:
I already removed the `onError` param from `useSubscription`